File: golang/prob2/main.go
package main

import "fmt"

func main(){

  fibOne := 1
  fibTwo := 2

  sum := 2

  for fibTwo <= 4000000 {
    fibTwo = fibTwo + fibOne
    fibOne = fibTwo - fibOne

    if fibTwo % 2 == 0 {
      sum += fibTwo
    }
  }

  fmt.Printf("The answer is: %v\n", sum)
  return

}

File: elixir/prob2/main.exs
defmodule EulerTwo do
  def main() do
    even_fibonacci_sum(1, 2, 2, 4000000)
  end

  def even_fibonacci_sum(fib_one, fib_two, sum, end_value) do
    fib_three = fib_one + fib_two
    cond do
      fib_three > end_value ->
        sum
      true ->
        sum = if even?(fib_three) do
          sum + fib_three
        else
          sum
        end

        even_fibonacci_sum(fib_two, fib_three, sum, end_value)
    end
  end

  def even?(value) do
    cond do
      rem(value, 2) == 0 ->
        true
      true ->
        false
    end
  end
end
